aimmdb/postprocessing/operations.py

```diff
@@ -165,7 +168,108 @@ def __call__(self, *inps):
             )
 
 
-class AverageData(MultiOperator):
+class MimoOperator(Operator):
+    """
+    Multiple input multiple output (MIMO) operator class.
+    Specialized operator class which takes an arbitrary number of inputs
+    and returns a list with an equal number of outputs.
+    All inputs must be of instance :class:`DataFrameClient` or `dict`.
+
+    Particularly, the operator object's ``__call__`` method can be executed on
+    an arbitrary number of :class:`DataFrameClient` or `dict` objects. For each
+    object passed some operation will be performed and the result will be returned
+    as a `dict` with keys "data" and "metatdata".
+    """
+
+    def _process_metadata(self, *metadata):
+        """Processing of the metadata dictionary objects. Takes arbitrary number
+        of :class:`dict` objects as input and returns a new dictionary for each input
+        with the following:
+
+            1. metadata["_tiled"]["uid"] is replaced with a new uid string.
+            2. metadata["post_processing"] is created with keys that indicate
+               the current state of the class. This includes a "parent" key
+               with the uid of the specific object that was operated on, and
+               a "relatives" key with the uids of all operator inputs.
+
+        Parameters
+        ----------
+        metadata : dict
+            The metadata dictionaries accessed via ``df_client.metadata``.
+
+        Returns
+        -------
+        list of dict
+            List of new metadata object for each post-processed child.
+        """
+
+        dt = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        input_uids = [md["_tiled"]["uid"] for md in metadata]
+        return [
+            {
+                "_tiled": {"uid": uid()},  # Assign a new uid
```

Comment:
Again this was my mistake, we don't have to assign a `uid`. Tiled will do this for us.
@jmaruland can you quickly confirm that this is true? Thanks!

Reply:
@x94carbone for the time being do we want to just leave this in until it becomes more clear how the operators should interface with tiled for the uids?